### PR TITLE
CAMEL-11019- message history formatting via globaloptions

### PR DESCRIPTION
**Pull request by onders86**
i tried to help about the suggestion in the issue.

### CODE REVIEW

File: camel-core/src/main/java/org/apache/camel/util/MessageHelper.java [hunk 1]
```diff
@@ -546,7 +546,10 @@ public static String doDumpMessageHistoryStacktrace(Exchange exchange, ExchangeF
         sb.append("\n");
         sb.append("Message History\n");
         sb.append("---------------------------------------------------------------------------------------------------------------------------------------\n");
-        sb.append(String.format(MESSAGE_HISTORY_HEADER, "RouteId", "ProcessorId", "Processor", "Elapsed (ms)"));
+        String goMessageHistoryHeaeder = exchange.getContext().getGlobalOption(Exchange.MESSAGE_HISTORY_HEADER);
+        sb.append(String.format(
+                         goMessageHistoryHeaeder == null ? MESSAGE_HISTORY_HEADER : goMessageHistoryHeaeder,
+                         "RouteId", "ProcessorId", "Processor", "Elapsed (ms)"));
```
Comment (zregvart) [inline camel-core/src/main/java/org/apache/camel/util/MessageHelper.java]: I would prefer to write this as:

    final CamelContext context = exchange.getContext();
    final String historyHeaderFormat = Optional.ofNullable(context.getGlobalOption(Exchange.MESSAGE_HISTORY_HEADER).orElse(MESSAGE_HISTORY_HEADER);
    sb.append(String.format(historyHeaderFormat, "RouteId", "ProcessorId", "Processor", "Elapsed (ms)"));

This way it's a bit cleaner.

File: camel-core/src/main/java/org/apache/camel/util/MessageHelper.java [hunk 2]
```diff
@@ -562,7 +565,10 @@ public static String doDumpMessageHistoryStacktrace(Exchange exchange, ExchangeF
             elapsed = new StopWatch(created).stop();
         }
 
-        sb.append(String.format(MESSAGE_HISTORY_OUTPUT, routeId, id, label, elapsed));
+        String goMessageHistoryOutput = exchange.getContext().getGlobalOption(Exchange.MESSAGE_HISTORY_OUTPUT);
+        sb.append(String.format(
+                        goMessageHistoryOutput == null ? MESSAGE_HISTORY_OUTPUT : goMessageHistoryOutput,
+                        routeId, id, label, elapsed));
```
Comment (zregvart) [inline camel-core/src/main/java/org/apache/camel/util/MessageHelper.java]: The same as above, I would prefer to use Optional

File: camel-core/src/main/java/org/apache/camel/Exchange.java [hunk 3]
```diff
@@ -169,6 +169,8 @@
     String MAXIMUM_TRANSFORMER_CACHE_SIZE = "CamelMaximumTransformerCacheSize";
     String MAXIMUM_VALIDATOR_CACHE_SIZE = "CamelMaximumValidatorCacheSize";
     String MESSAGE_HISTORY             = "CamelMessageHistory";
+    String MESSAGE_HISTORY_HEADER      = "CamelMessageHistoryHeader";
+    String MESSAGE_HISTORY_OUTPUT      = "CamelMessageHistoryOutput";
```
Comment (zregvart) [inline camel-core/src/main/java/org/apache/camel/Exchange.java]: I'm not entirely sure that we should use that wording, could `String MESSAGE_HISTORY_HEADER_FORMAT = "CamelMessageHistoryHeaderFormat"` and `String MESSAGE_HISTORY_OUTPUT_FORMAT = "CamelMessageHistoryOutputFormat"` be better, so we don't mix them with message headers?

Comment (onders86) [inline camel-core/src/main/java/org/apache/camel/Exchange.java]: yes, agreed. i will updated as your suggesstion.

File: camel-core/src/main/java/org/apache/camel/util/MessageHelper.java (hunk 1 above)
Comment (onders86) [inline camel-core/src/main/java/org/apache/camel/util/MessageHelper.java]: yes, it looks cleaner. doesn't it drop out the change of this fix would go into 2.17.x releases?

**Comment by zregvart** [comment]
@onders86 thanks for your contribution 👍 the changes have been squashed and merged into master. I've changed the title of the issue and the commit message to better reflect the change. You can close the pull request, thanks 🥇

**Comment by onders86** [comment]
thanks.closing

**Pull request closed without merging**